### PR TITLE
As branch-Update contrastive

### PR DESCRIPTION
**Pull request by ashkanfard132**

### CODE REVIEW

File: contrastive_and_svm.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from loss import SVMHingeLoss, SupConLoss
+from tqdm import tqdm
+from loss import melanoma_loss
+import wandb
+
```
Comment (monezamipoor) [inline contrastive_and_svm.py:8]: You see, this function is run per epoch. Meaning that it is defining the loss function and model and everything else once per epoch. For even contrastive learning you're using, this is not going to work.

File: main.py [hunk 2]
```diff
@@ -421,19 +335,9 @@ def train_batch(melanomamodel, images, labels, epoch):
         preds = melanomamodel.model(images)
 
         loss_function = melanomamodel.opt['model'].get('loss_function', 'bce').lower()
-        if loss_function == 'bce':
```
Comment (monezamipoor) [inline main.py:424]: Why defining every loss function again why we have a file for loss function selection? And also this is done per epoch. The correct way is for it to be defined once in the base MelanomaTrainer class and use these objects in the main training loop.

**Comment by ashkanfard132** [comment]
Ok I will fix it

File: main.py [hunk 3]
```diff
@@ -447,88 +351,40 @@ def validate(m, val_loader, epoch=1):
...
+        loop = tqdm(val_loader, desc="[Val]")
+        for imgs, lbls in loop:
+            if isinstance(imgs, (list, tuple)):
+                imgs = imgs[0]
+            imgs, lbls = imgs.to(device), lbls.to(device)
+
+            preds = m.model(imgs)
+            probs = torch.sigmoid(preds)
+            loss = bce_crit(preds, lbls.float())
+            total_loss += loss.item()
```
Comment (monezamipoor) [inline main.py:373]: This preds is logits, we can't run evaluations based on logits.
